pandas/io/excel/_base.py

```diff
@@ -903,13 +919,51 @@ def __init__(
     ):
         if engine is None:
             engine = "xlrd"
+
+            # Determine ext and use odf for ods stream/file
             if isinstance(path_or_buffer, (BufferedIOBase, RawIOBase)):
+                ext = None
                 if _is_ods_stream(path_or_buffer):
                     engine = "odf"
             else:
                 ext = os.path.splitext(str(path_or_buffer))[-1]
                 if ext == ".ods":
                     engine = "odf"
+
+            # GH 35029 - Default to openpyxl if xlrd is not installed for non-xls
+            if (
+                engine == "xlrd"
+                and ext != ".xls"
+                and import_optional_dependency(
+                    "xlrd", raise_on_missing=False, on_version="ignore"
+                )
+                is None
+            ):
+                engine = "openpyxl"
+
+            # GH 35029 - Don't warn with xls files as only xlrd can read them
+            if engine == "xlrd" and ext != ".xls":
+                import inspect
+
+                caller = inspect.stack()[1]
+                if (
+                    caller.filename.endswith("pandas/io/excel/_base.py")
+                    and caller.function == "read_excel"
+                ):
+                    stacklevel = 4
+                else:
+                    stacklevel = 2
+                warnings.warn(
+                    "The default argument engine=None is deprecated. Using None "
+                    "defaults to the xlrd engine which is no longer maintained, "
+                    "and is not supported when using pandas with python >= 3.9. "
+                    "The default value will be 'openpyxl' in a future version of "
+                    "pandas, although xlrd will continue to be allowed for the "
+                    "indefinite future. Either install openpyxl and specify it as "
+                    "the engine or specify 'xlrd' to suppress this warning.",
+                    DeprecationWarning,
```

Comment:
why are we not just raising a FutureWarning? sure this might throw lots of warnings in user code but that is exactly where this needs to be changed and *not* library code (where a DeprecationWarning would be useful)

Reply:
Fore sure read_excel will also be used in library code

Reply:
yes it will be used in library code
but the vast majority would be in use code